src/ProjectNeon/Assets/Features/Effects/EffectOnAttacker.cs

```diff
@@ -0,0 +1,30 @@
+﻿using UnityEngine;
+using System.Collections;
+
+/**
+ * Effects that applies to the attacker on an attack event
+ */
+public class EffectOnAttacker : Effect
+{
+    private Effect effect;
+    private Member performer;
+    private Target effectTarget;
+
+    public EffectOnAttacker(Effect damage)
+    {
+        this.effect = damage;
+        BattleEvent.Subscribe<Attack>(attack => Execute(attack), this);
+    }
+
+    void Effect.Apply(Member source, Target target)
+    {
+        this.performer = source;
+        this.effectTarget = target;
+    }
+
+    void Execute(Attack attack)
+    {
+        this.effect.Apply(this.performer, new MemberAsTarget(attack.Attacker));
```

Comment:
Also, not correctly scoped. You must interrogate the `Attack` event to ensure that the member being attacked is the member with this effect.